pkg/api/api.go

```diff
@@ -240,6 +252,54 @@ func requestPostageBatchId(r *http.Request) ([]byte, error) {
 	return nil, errInvalidPostageBatch
 }
 
+type authKeyResponse struct {
+	Key string `json:"key"`
+}
+
+type roleRequest struct {
+	Role string `json:"role"`
+}
+
+func (s *server) authHandler(w http.ResponseWriter, r *http.Request) {
+	user, pass, ok := r.BasicAuth()
+
+	if !ok || !s.auth.Authorize(user, pass) {
+		s.logger.Debug("unauthorized")
```

Comment:
These log messages would be better with more information. Like with a prefix indicating that it is from the api and which handler.